app/src/main/java/com/example/tasky/common/data/util/APIUtilFunctions.kt

```diff
@@ -0,0 +1,34 @@
+package com.example.tasky.common.data.util
+
+import com.google.gson.Gson
+import retrofit2.HttpException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+
+@Singleton
+open class APIUtilFunctions @Inject constructor() {
+    fun parseErrorResponse(errorBody: String?): String {
+        return if (errorBody != null) {
+            try {
+                val apiErrorResponse = Gson().fromJson(errorBody, ApiErrorResponse::class.java)
+                apiErrorResponse.message
+            } catch (e: Exception) {
+                "$e"
+            }
```

Comment:
In general, you don't really need to parse the error body from the response to show it to the user. I would rather use the response HTTP code to determine which error to show. Using the error body won't allow you to localize your error messages, since you always get it as English text back. 
Usually, it's very clear based on the HTTP status what went wrong (or what to show to the user at least). Some examples:
/register + 409 -> Email already exists
/login + 401 -> Invalid credentials
413 -> Uploaded file was too large
500..599 -> Server side error, just display a generic error message (`Oops, something went wrong, please try again`)

Reply:
very good point and I did not think about this: **"Using the error body won't allow you to localize your error messages, since you always get it as English text back."**